### PR TITLE
Added list_prefix function

### PR DESCRIPTION
**Pull request by conorw8**
Added a ```list_prefix``` function that takes in an s3 url which mimics ```os.listdir()``` which returns a list of objects within the given url. Two optional args ```filter_files``` and ```filter_prefixes``` allows the user to specify to return only the files or prefixes within the given url.

Additionally, this package has been converted to publish to the DroneBase-ML private PyPi repo.

### CODE REVIEW

File: .github/workflows/publish.yaml [hunk 1]
```diff
@@ -2,12 +2,11 @@ name: Publish Python Package
 
 on:
   push:
-    branches:
-    - master
+    branches: [ main ]
```
Comment (jonlwowski012) [inline .github/workflows/publish.yaml:5]: Should this be master?

File: setup.py [hunk 2]
```diff
@@ -15,7 +15,7 @@
 if __name__ == "__main__":
     setup(
         name="ml-dronebase-data-utils",
-        version="0.0.5",
+        version="0.0.1",
```
Comment (jonlwowski012) [inline setup.py:18]: Is the version correct?

**Review by jonlwowski012 — APPROVED** [review]

File: setup.py (hunk 2 above)
Comment (conorw8) [reply setup.py:18]: Yes since I'm moving it to the private repo, it's like its starting over again.

File: .github/workflows/publish.yaml (hunk 1 above)
Comment (conorw8) [reply .github/workflows/publish.yaml:5]: Yep I need to fix that.

**Pull request merged**